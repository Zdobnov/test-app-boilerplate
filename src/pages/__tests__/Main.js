import React from 'react';
import { shallow } from 'enzyme';

import Main from '../Main';

describe('Main page', () => {
  const defaultProps = {};
  const renderComponent = props =>
    shallow(
      <Main
        {...defaultProps}
        {...props}
      />
    );

  it('should render component', () => {
    const component = renderComponent();

    expect(component).toMatchSnapshot();
  });
});
