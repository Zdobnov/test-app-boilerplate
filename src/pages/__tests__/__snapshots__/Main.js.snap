// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Main page should render component 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <Main />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": "I am Main page",
    },
    "ref": null,
    "rendered": "I am Main page",
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": "I am Main page",
      },
      "ref": null,
      "rendered": "I am Main page",
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
  },
}
`;
